src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -388,12 +396,20 @@ void IList.Remove(object value)
 
         void ICollection.CopyTo(Array array, int index)
         {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
-            if (array.Rank != 1)
-                throw new ArgumentException(nameof(array));
+            if ((array != null) && (array.Rank != 1))
+            {
+                throw new ArgumentException(); // Arg_RankMultiDimNotSupported
+            }
 
-            Array.Copy(_items, 0, array, index, _size);
+            try
+            {
+                // Array.Copy will check for null.
```

Comment:
What's the point of moving the check for null to `Array.Copy`? You have inserted a check for null above anyways: `(array != null) && (array.Rank != 1)`, so we're not gaining any speed here. It seems like it would be best just to keep that check as it was.